src/analysis/fourier/poisson_summation.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2022 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import analysis.fourier.add_circle
+import measure_theory.integral.exp_decay
+import analysis.special_functions.gaussian
+
+/-!
+# Poisson's summation formula
+
+We prove Poisson's summation formula `∑ (n : ℤ), f n = ∑ (n : ℤ), g n`, where `g` is the Fourier
+transform of `f`, under the following hypotheses:
+* `f` and `g` are continuous functions `ℝ → ℂ`.
+* `∀ t:ℝ, g t = ∫ x:ℝ, complex.exp (-2 * π * I * t * x) * f x`.
+* `f` is integrable.
+* The sum `∑ (n:ℤ), g n` is convergent.
+* For all compacts `K ⊂ ℝ`, the sum `∑ (n:ℤ), sup { ‖f(x + n)‖ | x ∈ K }` is convergent.
+
+The last three conditions follow if we assume both `f` and `g` have exponential decay -- this will
+be added in future.
```

Comment:
exponential decay is way too strong here. Why not do it for functions in the Schwartz space, which is still too strong but more natural?

Reply:
Happy to. In fact I think the proofs will go through for any continuous function with $O(|x|^{-(1 + \epsilon)})$ decay for some $\epsilon > 0$, which is considerably more general still. (Is there a good name for this space?)